### ISSUE TITLE
Backfill existing work versions

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#2526]
I believe this is in published_at in work_versions.

See https://github.com/sul-dlss/happy-heron/pull/2408#discussion_r946014784

**Comment by jcoyne** [comment]
@amyehodge can you say why we want to do this work?

**Comment by amyehodge** [comment]
@jcoyne Apologies for being so cryptic about this ticket.

Because the data hasn't been backfilled into this database field, when we run item reports, if we filter on "Date last deposited" for a time period before this field was created, we don't get any results, even though we know items were modified during that time period. This leads to misleading results when we do the reports, so it would be good to have that data filled in for all the existing items.

**Comment by jcoyne** [comment]
@amyehodge and where should we get the data to fill in this field?

**Comment by amyehodge** [comment]
@jcoyne It seems to me that the information about when objects were modified/updated exists in the history tables, and we are now producing recent activity tables, so we must have it stored somewhere. But perhaps I'm confused about what is being put in this column for the ones that have data.

**Comment by jcoyne** [comment]
I was able to update 1249 of the 5538 works that were in the deposit state by looking for a `deposit_complete` event and using that as the time.

```ruby
count = 0
Work.all.joins(:head).where('work_versions.published_at' => nil,
    'work_versions.state' => 'deposited').find_each do |work|
  event = work.events.find_by(event_type: 'deposit_complete')
  next unless event
  work.head.published_at = event.updated_at
  work.head.save
  count +=1
end
count
```

**Comment by amyehodge** [comment]
@jcoyne That's an improvement. It's still rather confusing to see so many items that are in the "deposited" state but that don't have a "date last deposited" in the report. I still see 4289 of these. 

![Screen Shot 2022-09-08 at 4 52 30 PM](https://user-images.githubusercontent.com/5558511/189223920-fa9eb801-cb35-4fde-86e1-cad8b1577c76.png)

**Comment by jcoyne** [comment]
@amyehodge I agree, but where do we get the data for these. There's no useful events that we can use. They were perhaps deposited before we added these events or they migrated from Hydrus.

**Comment by amyehodge** [comment]
@jcoyne Let me consult with the other users of this report to see how they feel about this. I think our two choices here are 1) this is fine and we'll leave it like this or 2) we decide this is too confusing to have incomplete and would prefer not to have this option. I'll let you know.

**Comment by amyehodge** [comment]
@jcoyne I've talked to @hannahfrost and @andrewjbtw and we will leave it like this, but I'm going to file a new ticket to request that this filter option be removed.

**Comment by amyehodge** [comment]
Use the accession timestamp from the workflow service in Argo or the last accessioned date, which we think are the same.

**Comment by jcoyne** [comment]
First I get the list of these items:
```
 data = Work.all.joins(:head).where('work_versions.published_at' => nil,
     'work_versions.state' => 'deposited').pluck(:druid, :head_id)
 =>
[["druid:bs961xz3901", 78],
...
 CSV.open('missing_published.csv', 'wb') do |csv|
   data.each do |row|
      csv << row
   end
 end
```

And then on the workflow-server-rails server, get the dates:
```
 arr_of_rows = CSV.read("/opt/app/workflow/missing_published.csv")
 =>
[["druid:bs961xz3901", "78"],
...
 arr_of_rows.count
 => 4281
 CSV.open("missing_published_with_times.csv", "wb") do |csv|
   arr_of_rows.each do |row|
     date = WorkflowStep.where(druid: row.first,workflow: 'accessionWF', process: 'end-accession').order('
version').last.completed_at.iso8601
     csv << [row.first, row.second, date]
   end
 end
```

Then finally back on h2, we update the dates
```
arr_of_rows = CSV.read("missing_published_with_times.csv")
 =>
[["druid:bs961xz3901", "78", "2020-05-11T20:48:38Z"],
...
 arr_of_rows.each do |row|
   work_version = WorkVersion.find(row.second)
   work_version.published_at = row.third
   work_version.save!
end
```

**Comment by amyehodge** [comment]
Looks great. Thank you @jcoyne !
